### DISCUSSION TITLE
feat: Prompt management  with langfuse

### DISCUSSION THREAD

**Discussion by flefevre** [open-webui/open-webui#5915] [Ideas]
In order to improve the prompt life cycle, it could be good to be able to match the open webui prompts saved in the user interface to a specific prompt management system such as the one of langfuse

By typing / you could retrieve the different prompt managed by langfuse 

https://langfuse.com/docs/prompts/get-started

**Comment by flefevre** [comment]
The idea is to rely on prompt that are not directly store in open webui but on prompts that are managed by langfuse to ensure their full life cycle management.

Could we iterate on this idea?

The '/' command allow for instance to load prompts defined in langfuse system.

Perhaps the workspace should have specific parameters to define the langflow URL, secret key and the authorized prompts

https://langfuse.com/docs/prompts/example-langchain

**Comment by flefevre** [comment]
I have tried to draft an enhancement ticket to help you to gather the main ideas.


💡 Add Gateway to Langfuse for Listing Managed Prompts in Open We

# Summary

This proposal suggests adding a gateway in Open WebUI to connect with Langfuse. By creating this integration, prompts managed in Langfuse could be directly accessible from within Open WebUI, potentially in the prompt suggestions or model extensions.

# Problem Statement

Open WebUI currently lacks an integration with external prompt management systems like Langfuse. Integrating Langfuse would provide a way to:

Centralize Prompt Management: Allow Open WebUI users to see and reuse prompts written and stored in Langfuse, without having to duplicate them in the interface.

Seamless Access to Prompt Libraries: Accessing prompts stored in Langfuse directly in Open WebUI could simplify the workflow, especially when using multiple prompts or models.


# Proposed Solution

Implement a gateway connection to Langfuse through the Langfuse API. The connection would enable Open WebUI to:

1. List Prompts Stored in Langfuse: Display prompts from Langfuse in the Open WebUI prompt suggestions or model extensions interface.


2. Provide Contextual Prompt Suggestions: By linking with Langfuse, users could browse and select from existing prompts without needing to leave the Open WebUI environment.



# Implementation Suggestions

1. API Connection: Use the Langfuse API to fetch and list prompts within Open WebUI. A minimal setup is needed, following Langfuse's documentation for prompt management.


2. UI Integration: Add a simple UI component where prompts from Langfuse can be viewed and selected, such as in the prompt suggestions or model extensions sections.


3. Documentation: Include setup instructions for users to link their Langfuse account with Open WebUI.



# Benefits

Efficient Prompt Management: Users would be able to centralize and organize prompts in Langfuse while using Open WebUI.

Enhanced Usability: Accessing prompts directly from Langfuse within Open WebUI would streamline workflows for users who manage large prompt libraries.

Modularity: This feature is focused on connection only, allowing users to choose Langfuse as a prompt management layer without significant changes to Open WebUI.

**Comment by flefevre** [comment]
I have just noticed Langfuse add documentation to have trace of call from open webui with pipelines framework 
https://langfuse.com/docs/integrations/openwebui

Nevertheless it do not offer any service on prompt management system.

**Comment by flefevre** [comment]
Already in the roadmap 
https://github.com/open-webui/open-webui/issues/5962

**Reply by flefevre** [reply]
Langfuse team propose to use their REST API to make the link between open webui and langfuse: https://github.com/orgs/langfuse/discussions/4091

**Comment by mkagit** [comment]
Any news about it?
